agave/filters.py

```diff
@@ -20,3 +23,68 @@ def generic_query(query: QueryParams) -> Q:
     if 'count' in fields:
         del fields['count']
     return filters & Q(**fields)
+
+
+def get_by_id(cls, id: str):
+    try:
+        id_obj = cls.model.objects.get(id=id)
+    except Exception as exc:
+        if isinstance(exc, DoesNotExist):
+            raise
+        id_obj = cls.model.get_by(id=id)
+        if not id_obj:
+            raise
+    return id_obj
+
+
+def get_by_id_and_user(cls, id: str):
+    try:
+        id_query = Q(id=id)
+        id_query = id_query & Q(user_id=cls.current_user_id)
+        id_obj = cls.model.objects.get(id_query)
+    except Exception as exc:
+        if isinstance(exc, DoesNotExist):
+            raise
+        id_obj = cls.model.query.filter(
+            id=id,
+            user=cls.current_user.o_id,  # It can only query the numeric index
+        ).first()
+        if not id_obj:
+            raise
+    return id_obj
+
+
+def generic_query_redis(query: QueryParams, **kwargs) -> Dict[str, Any]:
```

Comment:
where's `generic_query_redis`? There has to be symmetry

Reply:
should be `generic_redit_query` and `generic_mongo_query`

Reply:
Done!